onnxruntime/core/providers/partitioning_utils.cc

```diff
@@ -51,6 +51,38 @@ std::string NodeGroupDebugString(const Container& group, bool show_all = false)
 }
 #endif
 
+/**
+Create partition node groups.
+
+A partition node group (a.k.a. a group) contains supported nodes that will run in a partition.
+
+All nodes in a group can be run together. This means that two nodes with an intervening unsupported node cannot be in
+the same group. On the other hand, nodes within the same group do not necessarily have to be connected.
+
+The partitioning algorithm attempts to form the largest possible groups in a greedy fashion. It is a variant of Kahn's
+topological sort algorithm that forms the group(s) as it goes.
+
+Conceptually, we consider nodes in a sequence of waves starting from the root nodes. One wave produces at most one
+group. A wave flows over nodes in topological order, adding supported nodes to the current group, and stops at the
+border of the current group. The next wave starts where the previous wave stopped.
+
+In the topological sort, we maintain a set of nodes that have no unprocessed inputs from which we select the next node
+to process.
+
+When selecting the next node to process, we first take:
+- a supported node (which will be part of the group)
+- an unsupported node that is not an output of any node in the group
```

Comment:
nit,  an unsupported node that does not consume an output of any node in the group